src/realm/object-store/c_api/socket_provider.cpp

```diff
@@ -15,8 +15,7 @@ struct CAPITimer : sync::SyncSocketProvider::Timer {
               realm_sync_socket_create_timer_func_t create_timer_func,
               realm_sync_socket_timer_canceled_func_t cancel_timer_func,
               realm_sync_socket_timer_free_func_t free_timer_func)
-        : m_handler(handler)
```

Comment:
what prompted this change?

Reply:
We were allowing the CAPITimer to delete the handler, but really it should have been "owned" by the implementation and freed when the complete/cancelled function was called, similar to the post and write_async callbacks.

Reply:
I think the idea was that if none of the functions is called and the timer is deallocated, it then deallocates the handler too. Is that still possible?